06_deploy_bootstrap_vm.sh

```diff
@@ -6,63 +6,25 @@ source ocp_install_env.sh
 source common.sh
 source utils.sh
 
-# FIXME this is configuring for the libvirt backend which is dev-only ref
-# https://github.com/openshift/installer/blob/master/docs/dev/libvirt-howto.md
-# We may need some additional steps from that doc in 02* and also to make the
-# qemu endpoint configurable?
 if [ ! -d ocp ]; then
     mkdir -p ocp
-    cat > ocp/install-config.yaml << EOF
+    cat > ocp/baremtal-install-config.yaml << EOF
 apiVersion: v1beta3
 baseDomain: ${BASE_DOMAIN}
 metadata:
   name: ${CLUSTER_NAME}
 platform:
-  libvirt: {}
+  baremetal: {}
 pullSecret: |
   ${PULL_SECRET}
 sshKey: |
   ${SSH_PUB_KEY}
 EOF
 fi
 
-
-$GOPATH/src/github.com/openshift/installer/bin/openshift-install --dir ocp --log-level=debug create ignition-configs
-
-# Here we can make any necessary changes to the ignition configs/manifests
-# they can later be sync'd back into the installer via a new baremetal target
-
-# Now re create the cluster (using the existing install-config and ignition-configs)
-# Since we set the replicas to null, we should only get the bootstrap VM
-# FIXME(shardy) this doesn't work, it creates the bootstrap and one master
-#$GOPATH/src/github.com/openshift/installer/bin/openshift-install --dir ocp --log-level=debug create cluster
-#exit 1
-
-# ... so for now lets create the bootstrap VM manually and use the generated ignition config
-# See https://coreos.com/os/docs/latest/booting-with-libvirt.html
-IGN_FILE="/var/lib/libvirt/images/${CLUSTER_NAME}-bootstrap.ign"
-sudo cp ocp/bootstrap.ign ${IGN_FILE}
-
-# Apply patches to bootstrap ignition
-apply_ignition_patches bootstrap "$IGN_FILE"
-
-LATEST_IMAGE=$(ls -ltr redhat-coreos-maipo-*-qemu.qcow2 | tail -n1 | awk '{print $9}')
-sudo cp $LATEST_IMAGE /var/lib/libvirt/images/${CLUSTER_NAME}-bootstrap.qcow2
-sudo qemu-img resize /var/lib/libvirt/images/${CLUSTER_NAME}-bootstrap.qcow2 50G
-sudo virt-install --connect qemu:///system \
-             --import \
-             --name ${CLUSTER_NAME}-bootstrap \
-             --ram 4096 --vcpus 4 \
-             --os-type=linux \
-             --os-variant=virtio26 \
-             --disk path=/var/lib/libvirt/images/${CLUSTER_NAME}-bootstrap.qcow2,format=qcow2,bus=virtio \
-             --vnc --noautoconsole \
-             --network bridge=baremetal --network bridge=brovc \
-             --print-xml > ocp/bootstrap-vm.xml
-sed -i 's|type="kvm"|type="kvm" xmlns:qemu="http://libvirt.org/schemas/domain/qemu/1.0"|' ocp/bootstrap-vm.xml
-sed -i "/<\/devices>/a <qemu:commandline>\n  <qemu:arg value='-fw_cfg'/>\n  <qemu:arg value='name=opt/com.coreos/config,file=${IGN_FILE}'/>\n</qemu:commandline>" ocp/bootstrap-vm.xml
-sudo virsh define ocp/bootstrap-vm.xml
-sudo virsh start ${CLUSTER_NAME}-bootstrap
+# "create cluster" only launches the bootstrap VM only for now
+cp ocp/baremtal-install-config.yaml ocp/install-config.yaml
+$GOPATH/src/github.com/metalkube/kni-installer/bin/kni-install --dir ocp --log-level=debug create cluster
```

Comment:
Interested to hear how far you got with this, e.g did modify things for the baremetal target so we only launch the bootstrap VM? (When I tested this setting replicas to null for masters still resulted in a single master getting deployed.)

Reply:
Yes, the baremetal platform driver so far only knows how to launch the bootstrap VM - adding support for provisioning the masters can come later

Reply:
Ack, as mentioned above I'm fuzzy on how we plan to do that, e.g should we build on the gophercloud ironic support to do it via terraform, or just consume the ignition-configs and build the nodes outside the installer.

Reply:
Captured some thoughts on master node provisioning here - https://github.com/metalkube/kni-installer/blob/master/docs/dev/baremetal.md